jetty-core/jetty-siwe/src/main/java/org/eclipse/jetty/security/siwe/EthereumAuthenticator.java

```diff
@@ -0,0 +1,772 @@
+
+        String maxMessageSize = authConfig.getParameter(MAX_MESSAGE_SIZE_PARAM);
+        if (maxMessageSize != null)
+            setMaxMessageSize(Integer.parseInt(maxMessageSize));
+
+        String logout = authConfig.getParameter(LOGOUT_REDIRECT_PARAM);
+        if (logout != null)
+            setLogoutRedirectPath(logout);
+
+        String error = authConfig.getParameter(ERROR_PAGE);
+        if (error != null)
+            setErrorPage(error);
+
+        String dispatch = authConfig.getParameter(DISPATCH_PARAM);
+        if (dispatch != null)
+            setDispatch(Boolean.parseBoolean(dispatch));
+
+        // If no LoginService is set we allow any user to log in.
+        if (authConfig.getLoginService() == null)
+        {
+            LoginService loginService = new AnyUserLoginService();
+            authConfig = new Configuration.Wrapper(authConfig)
+            {
+                @Override
+                public LoginService getLoginService()
+                {
+                    return loginService;
+                }
+            };
+        }
+
+        super.setConfiguration(authConfig);
+    }
+
+    @Override
+    public String getAuthenticationType()
+    {
+        return Authenticator.SIWE_AUTH;
+    }
+
+    public void setLoginPath(String loginPath)
+    {
+        if (loginPath == null)
+        {
+            LOG.warn("login path must not be null, defaulting to " + _loginPath);
+            loginPath = _loginPath;
+        }
+        else if (!loginPath.startsWith("/"))
+        {
+            LOG.warn("login path must start with /");
+            loginPath = "/" + loginPath;
+        }
+
+        _loginPath = loginPath;
+    }
+
+    public void setAuthPath(String authPath)
+    {
+        if (authPath == null)
+        {
+            authPath = _authPath;
+            LOG.warn("login path must not be null, defaulting to " + authPath);
+        }
+        else if (!authPath.startsWith("/"))
+        {
+            authPath = "/" + authPath;
+            LOG.warn("login path must start with /");
+        }
+
+        _authPath = authPath;
+    }
+
+    public void setNoncePath(String noncePath)
+    {
+        if (noncePath == null)
+        {
+            noncePath = _noncePath;
+            LOG.warn("login path must not be null, defaulting to " + noncePath);
+        }
+        else if (!noncePath.startsWith("/"))
+        {
+            noncePath = "/" + noncePath;
+            LOG.warn("login path must start with /");
+        }
+
+        _noncePath = noncePath;
+    }
+
+    public void setMaxMessageSize(int maxMessageSize)
+    {
+        _maxMessageSize = maxMessageSize;
+    }
+
+    public void setDispatch(boolean dispatch)
+    {
+        _dispatch = dispatch;
+    }
+
+    public void setLogoutRedirectPath(String logoutRedirectPath)
+    {
+        if (logoutRedirectPath == null)
+        {
+            LOG.warn("logout redirect path must not be null, defaulting to /");
```

Comment:
No warn, I'd stay at debug or at most info.

Reply:
removed this if branch altogether because logoutRedirectPath can be null.